libpod/container_internal.go

```diff
@@ -1307,9 +1307,7 @@ func (c *Container) stop(timeout uint) error {
 		c.lock.Unlock()
 	}
 
-	if err := c.ociRuntime.StopContainer(c, timeout, all); err != nil {
-		return err
-	}
+	stopErr := c.ociRuntime.StopContainer(c, timeout, all)
```

Comment:
Could you drop a comment here explaining why we must move the error check _below_ the potential `Lock()`.  It's easy to miss and since we can't test, we may silently regress.

Reply:
Done